src/components/space/charts/ActionsByTimeOfDayChart.js

```diff
@@ -38,17 +40,39 @@ const ActionsByTimeOfDayChart = () => {
   // third condition above is necessary: some actions are made by users NOT in the users list (e.g. user account deleted)
   // e.g. we retrieve 100 total actions and 10 users, but these 10 users have only made 90 actions
   // therefore, to avoid confusion: when all users are selected, show all actions
-  let actionsByTimeOfDay;
-  if (
+  const groupBy = (key, arr) =>
+    arr.reduce(
+      (acc, cur) => ({
+        ...acc,
+        [cur[key]]: cur[key] in acc ? acc[cur[key]].concat(cur) : [cur],
+      }),
+      {},
+    );
```

Comment:
I think you can create a new file in `utils`, like `array.js` where you add this function.